### PR TITLE
feat: Conversation detail link

### PR DESCRIPTION
**Pull request by sepans**
The type of this PR is: Feature

This PR resolves [PURCHASE-2047]

### Description

This PR links `ConversationDetail` screen to `Conversation` screen using `NavigatoriOS`

The first 2 commits are just moving files around to make the directory structure comply with [this RFC](https://github.com/artsy/eigen/issues/3497). There is nothing in it except for moving files and updating imports. You can skip to the [third commit](https://github.com/artsy/eigen/commit/21efe5268e1f0426ee11d1148c7c1511597e3ccc) for review if the directory structure is correct.

<details><summary>Screenshots</summary>

detail-withicon

</details>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.
- [x] Add detail icon

[PURCHASE-2047]: https://artsyproduct.atlassian.net/browse/PURCHASE-2047

### CODE REVIEW

**Review by zephraph — APPROVED** [review]
Looks good @sepans!

There's a lot to review here, but the commits helped. I'd recommend trying to break it down in the future though, just so it's easier to track down related changes.

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
LGTM 🎉 #squashongreen

**Review by jo-rs — APPROVED** [review]
Not sure if that svg should move to its own file; that seems to be the way its done in the rest of the repo. But, otherwise, looks great! ⭐️

File: src/lib/Scenes/Inbox/Components/Conversations/Preview/Attachment/PDFPreview.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ interface Props extends AttachmentProps {
 export const PDFPreview: React.FC<Props> = ({ attachment, onSelected }) => (
   <AttachmentPreview attachment={attachment} onSelected={onSelected}>
     <AttachmentContainer>
-      <Icon source={require("../../../../../../../images/pdf.png")} />
+      <Icon source={require("../../../../../../../../images/pdf.png")} />
```
[the remark this answers is not in the document]
Comment (jo-rs) [reply src/lib/Scenes/Inbox/Components/Conversations/Preview/Attachment/PDFPreview.tsx:36]: Where have you been all my life?

**Comment by sepans** [comment]
😓 I addressed the `@image` and `svg` comments but my simulator is broken at the time being and didn't want to push untested code so I will create another PR for that later. Thanks for the reviews 🙌

**Pull request merged**